zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/client/HttpCall.java

```diff
@@ -13,27 +13,44 @@
  */
 package zipkin2.elasticsearch.internal.client;
 
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.util.ByteBufferBackedInputStream;
 import com.linecorp.armeria.client.HttpClient;
 import com.linecorp.armeria.common.AggregatedHttpRequest;
 import com.linecorp.armeria.common.AggregatedHttpResponse;
 import com.linecorp.armeria.common.HttpData;
 import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
 import com.linecorp.armeria.common.HttpStatusClass;
 import com.linecorp.armeria.common.RequestContext;
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.ByteBufHolder;
 import io.netty.util.ReferenceCountUtil;
+import java.io.FileNotFoundException;
 import java.io.IOException;
-import java.nio.ByteBuffer;
+import java.io.InputStream;
 import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.Semaphore;
 import zipkin2.Call;
 import zipkin2.Callback;
 
+import static zipkin2.elasticsearch.internal.JsonReaders.enterPath;
+import static zipkin2.elasticsearch.internal.JsonSerializers.JSON_FACTORY;
+
 public final class HttpCall<V> extends Call.Base<V> {
 
-  public interface BodyConverter<V> {
-    V convert(ByteBuffer content) throws IOException;
+  public static abstract class BodyConverter<V> {
+    /** Most convert with {@link HttpData#toStringUtf8()} or {@link #toInputStream(HttpData)} */
+    public abstract V convert(HttpData content) throws IOException;
+
+    // TODO: once https://github.com/line/armeria/issues/1918 is done, switch back to an interface
+    /** Use this when you don't need a string or only need to read the response once. */
+    protected InputStream toInputStream(HttpData content) {
+      if (content instanceof ByteBufHolder) {
+        return new ByteBufferBackedInputStream(((ByteBufHolder) content).content().nioBuffer());
```

Comment:
`return new ByteBufInputStream(content.content())`

Without setting `true` it won't release, and it's the same semantics with slightly better performance.

Reply:
k